src/components/diagrams/diagram-pane.tsx

```diff
@@ -275,18 +326,29 @@ const useDisplayView = (studyUuid: UUID, currentNode: CurrentTreeNode, currentRo
                 }
             }
 
-            if (diagramState.svgType === DiagramType.VOLTAGE_LEVEL) {
-                return createVoltageLevelDiagramView(diagramState.id!, diagramState.state);
-            } else if (diagramState.svgType === DiagramType.SUBSTATION) {
-                return createSubstationDiagramView(diagramState.id!, diagramState.state);
-            } else if (diagramState.svgType === DiagramType.NETWORK_AREA_DIAGRAM) {
-                return createNetworkAreaDiagramView(diagramState.ids, diagramState.state, diagramState.depth);
+            if (partialDiagramView.svgType === DiagramType.VOLTAGE_LEVEL) {
+                return createVoltageLevelDiagramView(partialDiagramView.id!, partialDiagramView.state);
+            } else if (partialDiagramView.svgType === DiagramType.SUBSTATION) {
+                return createSubstationDiagramView(partialDiagramView.id!, partialDiagramView.state);
+            } else if (partialDiagramView.svgType === DiagramType.NAD_FROM_CONFIG) {
+                return createNetworkAreaDiagramFromConfigView(
+                    partialDiagramView.id!,
```

Comment:
by definition of "Partial DiagramView", id could be undefined here
if we know it can't be undefined here, change Partial to something different
otherwise, handle the undefined value

Reply:
Ok I just realised it's done the same way with other DiagramType... to do in another PR